### PR TITLE
Adding klicktipp subscribe and unsubscribe api

### PR DESCRIPTION
**Pull request by einhornimmond**
## 🍰 Pullrequest

# Klicktipp

## Subscribe
Subscribe current logged in user to gradido newsletter

### Request
`GET http://localhost/api/klicktipp_subscribe/[session_id]`
Parts symbolized by [] are optional
  - session_id: session will be searched in php session and GRADIDO_LOGIN cookie and if not found use this

### Response
Assuming: session is valid

```json
{
    "state": "success",
    "redirect_url": "<redirect url from klicktipp>"
}
````

## Unsubscribe
Unsubscribe current logged in user from gradido newsletter

### Request
`GET http://localhost/api/klicktipp_unsubscribe/[session_id]`
Parts symbolized by [] are optional
  - session_id: session will be searched in php session and GRADIDO_LOGIN cookie and if not found use this

### Response
Assuming: session is valid

```json
{
    "state": "success"
}
````

### CODE REVIEW

**Comment by ulfgebhardt** [comment]
Can you add the docu to the corresponding docu file?
Can you call subscribe on register?

File: community_server/src/Controller/AppRequestsController.php [hunk 1]
```diff
@@ -352,5 +355,77 @@ private function acquireAccessToken($session_id)
...
+        
+        $api_key =  Configure::read('KlickTipp_API_KEY');
+        $email_address = $user['email']; // Replace with the email address.
+        $fields = array ( // Use field_index to get all custom fields.
+          'fieldFirstName' => $user['first_name'],
+          'fieldLastName' => $user['last_name'],
+        );
+        
+
+        $connector = new KlicktippConnector();
```
Comment (ulfgebhardt) [inline community_server/src/Controller/AppRequestsController.php:379]: Where is the `KlicktippConnector` coming from?

File: community_server/src/Controller/AppRequestsController.php [hunk 2]
```diff
@@ -352,5 +355,77 @@ private function acquireAccessToken($session_id)
...
+        
+        /**
+        * Add subscriber. API Key required.
+        * 
+        * @param mixed $api_key
+        * @param mixed $email_address
+        * @param mixed $fields (optional)
+        * @param mixed $smsnumber (optional)
+        * 
+        * @return redirect url as defined in the subscription process used in the API Key.
```
Comment (ulfgebhardt) [inline community_server/src/Controller/AppRequestsController.php:389]: whats the `redirect_url`

File: community_server/src/Controller/AppRequestsController.php (hunk 1 above)
Comment (einhornimmond) [reply community_server/src/Controller/AppRequestsController.php:379]: It is the Klicktipp PHP-Wrapper

File: community_server/src/Controller/AppRequestsController.php (hunk 2 above)
Comment (einhornimmond) [reply community_server/src/Controller/AppRequestsController.php:389]: I don't know

**Comment by einhornimmond** [comment]
I have added subscribeClicktip but with default value false.

**Comment by ulfgebhardt** [comment]
Pushed to May to be handled with the register process

**Comment by Elweyn** [comment]
Has been transfered to Apollo.

**Pull request closed without merging**